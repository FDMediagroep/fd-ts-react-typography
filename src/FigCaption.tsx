import React, { PureComponent } from "react";
import { createGlobalStyle } from "styled-components";

export interface Props {
    caption?: string;
    className?: string;
    photographer?: string;
}

export default class FigCaption extends PureComponent<Props, any> {
    render() {
        return (
            <>
                <GlobalStyle/>
                <figcaption {...this.props} className={`fd-figcaption${this.props.className ? ` ${this.props.className}` : ''}`}>
                    <span className="caption">{this.props.caption}</span>
                    <span className="photographer">{this.props.photographer}</span>
                </figcaption>
            </>
        );
    }
}

const GlobalStyle = createGlobalStyle`
.fd-figcaption {
    &.article-inline {
        color: #4c4c4c;
        text-align: left;
        padding: 0.375rem 0;
        font-family: 'ProximaNovaRegular', sans-serif;
        line-height: 1.2em;
        font-size: 13px;
        @media only screen and (min-width: 641px) {
            font-size: 14px;
        }
        .photographer {
            margin-left: 5px;
            color: #677381;
        }
    }

    &.article-longread-inline {
        color: #8a8176;
        text-align: left;
        padding: 0.375rem 0;
        font-family: 'ProximaNovaRegular', sans-serif;
        line-height: 1.2em;
        font-size: 13px;
        @media only screen and (min-width: 641px) {
            font-size: 14px;
        }
        .photographer {
            margin-left: 5px;
            color: #677381;
        }
    }
}
`;

export {GlobalStyle as FigCaptionStyle};
