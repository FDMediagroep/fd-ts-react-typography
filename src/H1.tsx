import React, { PureComponent } from "react";
import { createGlobalStyle } from 'styled-components';

export default class H1 extends PureComponent<any, any> {
    render() {
        return (
            <>
                <H1Style/>
                <h1 {...this.props} className={`fd-h1${this.props.className ? ` ${this.props.className}` : ''}`}>{this.props.children}</h1>
            </>
        );
    }
}

export const H1Style = createGlobalStyle`
.fd-h1 {
    margin: 0;
    padding: 0;
    font-family: 'ArnhemProBold', serif;
    line-height: 1.08em;
    width: 88%;
    font-size: 1.6065rem;
    margin: 0;
    @media only screen and (min-width: 641px) {
        font-size: 2.8125rem;
    }
    @media only screen and (min-width: 1025px) {
        font-size: 3.25rem;
    }

    &.article-longread-persoonlijk {
        width: auto;
        font-size: 1.625rem;
        @media only screen and (min-width: 641px) {
            font-size: 3.4375rem;
        }
        @media only screen and (min-width: 1025px) {
            font-size: 4.375rem;
        }
    }

    &.article-longread {
        font-family: 'ConsortWebThin', serif;
        font-weight: normal;
        width: auto;
        font-size: 1.625rem;
        @media only screen and (min-width: 641px) {
            font-size: 3.4375rem;
        }
        @media only screen and (min-width: 1025px) {
            font-size: 4.375rem;
        }
    }

    &.opening-teaser {
        font-size: 1.5625rem;
        @media only screen and (min-width: 641px) {
            font-size: 1.8125rem;
        }
    }
    &.teaser {
        font-size: 1.375rem;
    }
    &.cover-teaser {
        font-size: 1.375rem;
        color: white;
    }
    &.square-teaser {
        font-size: 1.125rem;
        @media only screen and (min-width: 641px) {
            font-size: 1.1875rem;
        }
        @media only screen and (min-width: 1025px) {
            font-size: 1.25rem;
        }
    }
    &.service {
        font-family: 'ProximaNovaBold', Helvetica, Arial, sans-serif;
        color: #49a4a2;
        font-size: 1.875rem;
    }
    &.service-accordion {
        font-family: 'ProximaNovaRegular', Helvetica, Arial, sans-serif;
        font-size: 1rem;
        @media only screen and (min-width: 641px) {
            font-size: 1.0625rem;
        }
        a {
            display: block;
            color: #677381;
            cursor: pointer;
            font-weight: normal;
        }
    }
    &.my-news-aside {
        font-family: 'ProximaNovaBold', sans-serif;
        line-height: 1.2em;
        color: #49a4a2;
        font-size: 1rem;
        @media only screen and (min-width: 641px) {
            font-size: 1.125rem;
        }
    }
}`;
