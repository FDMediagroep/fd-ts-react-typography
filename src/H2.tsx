import React, { PureComponent } from "react";
import { createGlobalStyle } from "styled-components";

export default class H2 extends PureComponent<any, any> {
    render() {
        return (
            <>
                <GlobalStyle/>
                <h2 {...this.props} className={`fd-h2${this.props.className ? ` ${this.props.className}` : ''}`}>{this.props.children}</h2>
            </>
        );
    }
}

const GlobalStyle = createGlobalStyle`
.fd-h2 {
    margin: 0;
    padding: 0;
    &.article-subheader {
        max-width: 719px;
        padding-left: 0;
        padding-right: 0;
        margin: 2.25rem auto 0 auto;
        font-size: 1.375rem;
        font-family: 'ArnhemProBold', serif;
    }

    &.article-number-frame {
        font-weight: 400;
        color: #49a4a2;
        font-size: 3.125rem;
        font-family: 'ConsortWebRegular', serif;
        line-height: 1em;
    }
    &.article-longread-number-frame {
        font-weight: 400;
        color: #f05031;
        font-size: 3.125rem;
        font-family: 'ConsortWebRegular', serif;
        line-height: 1em;
    }

    &.article-link-block {
        color: #191919;
        font-size: 1.0625rem;
        font-family: 'ProximaNovaRegular', sans-serif;
        line-height: 1.2em;
    }
}
`;

export {GlobalStyle as H2Style};
