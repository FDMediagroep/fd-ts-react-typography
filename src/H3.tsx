import React, { PureComponent } from "react";
import { createGlobalStyle } from "styled-components";

export default class H3 extends PureComponent<any, any> {
    render() {
        return (
            <>
                <H3Style/>
                <h3 {...this.props} className={`fd-h3${this.props.className ? ` ${this.props.className}` : ''}`}>{this.props.children}</h3>
            </>
        );
    }
}

export const H3Style = createGlobalStyle`
.fd-h3 {
    margin: 0;
    padding: 0;
    font-family: 'ProximaNovaBold', sans-serif;
    box-sizing: border-box;
    display: block;
    font-size: 1rem;
    line-height: 1.1em;

    &.article-subheader {
        padding: 0;
        margin: 5rem auto 1.25rem auto;
        line-height: 1.15;
        font-family: 'ArnhemProBlond', serif;
        font-size: 1.875rem;
        @media only screen and (min-width: 1025) {
            font-size: 2.8125rem;
        }
    }
    &.article-longread-subheader {
        max-width: 719px;
        padding-left: 0;
        padding-right: 0;
        margin: 5rem auto 1.25rem auto;
        line-height: 1.15;
        font-family: 'ArnhemProBlond', serif;
        font-size: 1.875rem;
        @media only screen and (min-width: 1025) {
            font-size: 2.8125rem;
        }
    }

    &.article-inline {
        font-size: 1.25rem;
        font-family: 'ProximaNovaBold', sans-serif;
        color: #49a4a2;
        margin-bottom: 1rem;
    }

    &.article-longread-inline {
        font-size: 1.25rem;
        font-family: 'ProximaNovaBold', sans-serif;
        color: #f05031;
        margin-bottom: 1rem;
    }

    &.article-inline-stack-frame {
        font-size: 1.5rem;
        font-family: 'ProximaNovaBold', sans-serif;
        color: #4d494C;
        line-height: 2.025rem;
        margin: 0;
    }

    &.card {
        color: #49a4a2;
        font-family: 'ProximaNovaBold', sans-serif;
        line-height: 1.2em;
        @media only screen and (min-width: 641px) {
            font-size: 1.25rem;
        }
    }
}
`;
