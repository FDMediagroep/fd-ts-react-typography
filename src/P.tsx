import React, { PureComponent } from "react";
import { createGlobalStyle } from "styled-components";

export default class P extends PureComponent<any, any> {
    render() {
        return (
            <>
                <GlobalStyle/>
                <p {...this.props} className={`fd-paragraph${this.props.className ? ` ${this.props.className}` : ''}`}>{this.props.children}</p>
            </>
        );
    }
}

const GlobalStyle = createGlobalStyle`
    .fd-paragraph {
        margin: 0;
        padding: 0;

        &.article {
            font-family: 'ArnhemProBlond', serif;
            line-height: 1.5em;
            margin: 0 0 .5em 0;
            font-size: 1.125rem;
            @media only screen and (min-width: 641px) {
                font-size: 1.1875rem;
            }
            @media only screen and (min-width: 1025px) {
                font-size: 1.3125rem;
            }
            strong {
                font-family: 'ArnhemProBold', serif;
            }
            em {
                font-style: italic;
            }
            a {
                color: #49a4a2;
                cursor: pointer;
            }
        }
        &.article-longread {
            max-width: 719px;
            margin-left: auto;
            margin-right: auto;
            width: auto;
            margin-bottom: 0.625rem;
            font-size: 1.125rem;
            @media only screen and (min-width: 641px) {
                font-size: 1.1875rem;
            }
            @media only screen and (min-width: 1025px) {
                font-size: 1.3125rem;
            }
            font-family: 'ArnhemProBlond', serif;
            strong {
                font-family: 'ArnhemProBold', serif;
            }
            em {
                font-style: italic;
            }
            &:first-of-type::first-letter {
                font-size: 6rem;
                line-height: .8;
                margin: 0 1rem 0 -1rem;
                float: left;
                color: #f05031;
                text-transform: capitalize;
                font-family: 'ProximaNovaRegular', sans-serif;
            }
            a {
                color: #f05031;
                cursor: pointer;
            }
        }

        &.article-intro {
            color: #191919;
            width: 100%;
            margin: 0;
            font-size: 1.375rem;
            @media only screen and (min-width: 641px) {
                font-size: 1.5625rem;
            }
            @media only screen and (min-width: 1025px) {
                font-size: 1.75rem;
                width: 84%;
            }
            line-height: 1.2em;
            font-family: 'ArnhemProBlond', serif;
        }
        &.article-longread-intro {
            color: #8a8176;
            max-width: 719px;
            margin: 0 auto;
            width: auto;
            font-size: 1.125rem;
            @media only screen and (min-width: 641px) {
                font-size: 1.5625rem;
            }
            @media only screen and (min-width: 1025px) {
                font-size: 1.75rem;
            }
            line-height: 1.3em;
            text-align: center;
            font-family: 'ConsortWebBold', serif;
        }

        &.article-summary-p {
            display: inline-block;
            width: 100%;
            font-family: 'ProximaNovaRegular', sans-serif;
            font-size: 1.0625rem;
            line-height: 1.2;
            margin: 0 0 0.9375rem 0;
        }

        &.article-number-frame {
            color: #49a4a2;
            font-family: 'ProximaNovaRegular', sans-serif;
            font-size: 1.0625rem;
            line-height: 1.2;
            margin: 0 0 0.65625rem 0;
        }
        &.article-longread-number-frame {
            color: #f05031;
            font-family: 'ProximaNovaRegular', sans-serif;
            font-size: 1.0625rem;
            line-height: 1.2;
            margin: 0 0 0.65625rem 0;
        }

        &.article-text-frame {
            display: inline-block;
            width: 100%;
            font-size: 1.0625rem;
            font-family: 'ProximaNovaRegular', sans-serif;
            line-height: 1.434375rem;
            margin: 0 0 1.15622rem 0;
        }

        &.article-stack-frame {
            display: inline-block;
            font-size: 1.5rem;
            font-family: 'ProximaNovaBold', sans-serif;
            color: #49a4a2;
            line-height: 2.025rem;
            margin: 0;
        }
        &.article-longread-stack-frame {
            display: inline-block;
            font-size: 1.5rem;
            font-family: 'ProximaNovaBold', sans-serif;
            color: #f05031;
            line-height: 2.025rem;
            margin: 0;
        }

        &.article-link-block {
            color: #49a4a2;
            font-size: 1rem;
            font-family: 'ProximaNovaBold', sans-serif;
            line-height: 1.1em;
            @media only screen and (min-width: 641px) {
                font-size: 1.125rem;
            }
        }
        &.article-longread-link-block {
            color: #f05031;
            font-size: 1rem;
            font-family: 'ProximaNovaBold', sans-serif;
            line-height: 1.1em;
            @media only screen and (min-width: 641px) {
                font-size: 1.125rem;
            }
        }

        &.article-bullet-points {
            margin: 0 0 0.9375rem 0;
            color: #191919;
            font-size: 1.25rem;
            font-family: 'ProximaNovaRegular', sans-serif;
            line-height: 1.2;
        }

    }
`;

export {GlobalStyle as PStyle};
