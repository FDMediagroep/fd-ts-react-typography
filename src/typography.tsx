import H1, {H1Style} from './H1';
import H3, {H3Style} from './H3';

export {H1, H1Style};
export {H3, H3Style};
