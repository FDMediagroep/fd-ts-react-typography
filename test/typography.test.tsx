import TestRenderer from 'react-test-renderer';
import {H1, H3} from '../src/typography';
import React from 'react';

describe('Typography', () => {
    test('renders h1 correctly', () => {
        let h1 = TestRenderer.create(<H1>Default</H1>);
        expect(h1.toJSON()).toMatchSnapshot();
        h1 = TestRenderer.create(<H1 className="article-longread-persoonlijk">article-longread-persoonlijk</H1>);
        expect(h1.toJSON()).toMatchSnapshot();
    });

    test('renders h3 correctly', () => {
        let h3 = TestRenderer.create(<H3>Default</H3>);
        expect(h3.toJSON()).toMatchSnapshot();
        h3 = TestRenderer.create(<H3 className="card">card</H3>);
        expect(h3.toJSON()).toMatchSnapshot();
    });
});