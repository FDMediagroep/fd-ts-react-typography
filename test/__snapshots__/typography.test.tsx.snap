// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography renders h1 correctly 1`] = `
<h1
  className="fd-h1"
>
  Default
</h1>
`;

exports[`Typography renders h1 correctly 2`] = `
<h1
  className="fd-h1 article-longread-persoonlijk"
>
  article-longread-persoonlijk
</h1>
`;

exports[`Typography renders h3 correctly 1`] = `
<h3
  className="fd-h3"
>
  Default
</h3>
`;

exports[`Typography renders h3 correctly 2`] = `
<h3
  className="fd-h3 card"
>
  card
</h3>
`;
